pkg/operator/v1helpers/helpers.go

```diff
@@ -14,13 +14,44 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/diff"
 
 	"github.com/ghodss/yaml"
 
 	configv1 "github.com/openshift/api/config/v1"
 	operatorv1 "github.com/openshift/api/operator/v1"
 )
 
+var _ OperatorStatusUpdater = &LoggingStatusUpdater{}
+
+type LoggingStatusUpdater struct {
+	// cachedConditions store the last observed conditions to be returned in the
+	// GetOperatorState method in order to prevent excessive conditions update logging
+	cachedConditions []operatorv1.OperatorCondition
+}
+
+// LoggingStatusUpdater creates a new OperatorStatusUpdater that only logs the
+// requests to update current conditions.
+func NewLoggingStatusUpdater() *LoggingStatusUpdater {
+	return &LoggingStatusUpdater{}
+}
+
+func (c *LoggingStatusUpdater) GetOperatorState() (*operatorv1.OperatorSpec, *operatorv1.OperatorStatus, string, error) {
+	return &operatorv1.OperatorSpec{}, &operatorv1.OperatorStatus{
+		Conditions: c.cachedConditions,
+	}, "", nil
+}
+
+func (c *LoggingStatusUpdater) UpdateOperatorStatus(_ string, status *operatorv1.OperatorStatus) (*operatorv1.OperatorStatus, error) {
+	klog.V(2).Info("received a request to update status, diff: %s", diff.ObjectDiff(status, &operatorv1.OperatorStatus{Conditions: c.cachedConditions}))
```

Comment:
.oO(i wish we could inject the logger at initialization time) ... actually ... does that make sense, potentially just a logging function?

Reply:
`klog` is the logger all our operators use, I believe, and is heavily used in other controllers. I'd maybe do that in a follow-up if such a need arises.